src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1980,7 +1987,7 @@ internal bool CheckValueKind(BoundExpression expr, BindValueKind valueKind, Diag
                     return CheckEventValueKind((BoundEventAccess)expr, valueKind, diagnostics);
                 case BoundKind.DynamicMemberAccess:
                 case BoundKind.DynamicIndexerAccess:
-                    return true;
+                    return CheckDynamicValueKind(expr, valueKind, diagnostics);
```

Comment:
I was concerned about this change, but conflicting change has not been  merged in to the branch yet, so we should be fine.